### PR TITLE
misc_math: unit test: add tolerance to floating retvals

### LINKED ISSUE #1960: linear_interpolate unit tests fail on linux-i686
Probably a floating point precision thing.  All the values "match" on output.

### PR DESCRIPTION
**Pull request by mlyle**
Attempt to fix #1960.  Somewhat in the blind because I don't have linux-i686.  But I have pushed the branch to the Tau repository too, so we'll see if Jenkins spits out an i686 artifact.

@ekaszubski FYI.

### CODE REVIEW

**Comment by kubark42** [comment]
Looks reasonable. There's a fair amount of FPU action going on, so being at worst within 4 significant digits across the entire range is pretty good indication that the coverage is complete.

File: flight/tests/misc_math/unittest.cpp [hunk 1]
```diff
@@ -305,23 +305,24 @@ TEST_F(LinearInterpolate, ThrottleCurve1to1) {
   float const range_max = 1.0f;
   uint8_t const curve_numpts = 5;
   float const curve[5] = { 0.0f, 0.25f, 0.5f, 0.75f, 1.0f };
+  float eps = 0.000001f;
```
Comment (kubark42) [inline flight/tests/misc_math/unittest.cpp]: This could probably be several orders of magnitude more strict, but 1e-6 is fine.

Comment (mlyle) [inline flight/tests/misc_math/unittest.cpp]: Single precision floating point is 6 sigfigs of precision.  Arguably we should have one less zero.

Comment (kubark42) [inline flight/tests/misc_math/unittest.cpp]: Several means ```1```, [right](https://xkcd.com/1070/)? <grin>

**Comment by mlyle** [comment]
And this spat out of jenkins happily, so please merge?

**Comment by kubark42** [comment]
Yay, Jenkins!

**Pull request merged**